spinnaker/spinnaker_system/google_http_lb_upsert_server_test.py

```diff
@@ -0,0 +1,199 @@
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# See testable_service/integration_test.py and spinnaker_testing/spinnaker.py
+# for more details.
+#
+# The Http(s) test will be used to (more) thoroughly test the GCP L7 load
+# balancer Upsert and Delete commands, and will eventually be expanded to
+# include server group tests. This test operates on a L7 Lb with one server
+# group attached.
+
+# Standard python modules.
+from OpenSSL import crypto
+import copy
+import json
+import sys
+import time
+import uuid
+import google_http_upsert_test as upsert
+
+# citest modules.
+from citest.base import ExecutionContext
+import citest
+import citest.gcp_testing as gcp
+import citest.json_predicate as jp
+import citest.service_testing as st
+
+# Spinnaker modules.
+import spinnaker_testing as sk
+import spinnaker_testing.gate as gate
+
+
+SCOPES = [gcp.COMPUTE_READ_WRITE_SCOPE]
+
+class GoogleHttpLoadBalancerTest(st.AgentTestCase):
+  '''Test fixture for Http LB test.
+  '''
+
+  FIRST_CERT = ''
+  SECOND_CERT = ''
```

Comment:
are these certs only used by one test? What if you wrapped each of those tests so it creates the certs, runs the test, then deletes the cert (in a finally clause). Or if the cert needs to be valid over the future lifetime of the thing referencing it, then having the test add a deletion lambda after it creates the cert woudl also eliminate needing these class variables and access points spread across several different methods.